component/kafka/kafka.go

```diff
@@ -0,0 +1,73 @@
+package kafka
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/Shopify/sarama"
+	"github.com/beatlabs/patron/correlation"
+	"github.com/google/uuid"
+)
+
+// FailStrategy type definition.
+type FailStrategy int
+
+const (
+	// ExitStrategy does not commit failed message offsets and exits the application.
+	ExitStrategy FailStrategy = iota
+	// SkipStrategy commits the offset of messages that failed processing, and continues processing.
+	SkipStrategy
+)
+
+// BatchProcessorFunc definition of a batch async processor.
+type BatchProcessorFunc func(context.Context, []MessageWrapper) error
+
+// MessageWrapper interface for wrapping messages that are handled by the kafka component.
+type MessageWrapper interface {
```

Comment:
Check out the SQS consumer and the way it handles ACK. Inside of the ACK we also finish a tracing span. 
We should have similar interfaces here e.g. Message and Batch so that both component have the same look and feel.
Obvioulsy Kafka does not need a NACK because it exists by default.

Reply:
Off topic, is it on purpose that the `NACK()` in SQS does `SpanSuccess`? https://github.com/beatlabs/patron/blob/master/component/sqs/message.go#L93

Reply:
Yes, the span failure is only used when a message could not be deleted from SQS.
Span success means just that the processing succeeded.
Thing about the following scenario for example:
I got a message but it is not time to process it because I wait for something else to happen first.
So the NACK is telling SQS leave the message in the queue for later processing and the span is a success as an operation.

Reply:
Tracing handled in the consumer logic. 👍